### PR TITLE
Move editableInPropertyPage to Core and Update Observable Import

### PR DESCRIPTION
**Pull request by alexchuber**
Move editableInPropertyPage from editor to core package so that block libraries don't need to have a dependency on editor.

### CODE REVIEW

File: packages/core/src/editorUtils/editableInPropertyPage.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import type { Observable } from "@babylonjs/core";
+import type { Observable } from "@babylonjs/core/Misc/observable";
 
 /**
  * Enum defining the type of properties that can be edited in the property pages in the node editor
  */
-export const enum PropertyTypeForEdition {
+export enum PropertyTypeForEdition {
```
Comment (alexchuber) [inline packages/core/src/editorUtils/editableInPropertyPage.ts:6]: The TS option verbatimModuleSyntax = true does not agree with const enum exports?

File: packages/demo/src/configuration/blocks/inputs/webCamInputBlock.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+    ConnectionPointType,
+    type SmartFilter,
+    InputBlock,
+    createStrongRef,
     editableInPropertyPage,
     PropertyTypeForEdition,
     type IEditablePropertyListOption,
-} from "@babylonjs/smart-filters-editor";
+} from "@babylonjs/smart-filters";
+import type { InitializationData } from "core/src/smartFilter";
```
Comment (AmoebaChant) [inline packages/demo/src/configuration/blocks/inputs/webCamInputBlock.ts:12]: VSCode picked the direct path, could you pelase update to @bablyonjs/core?

**Review by AmoebaChant — CHANGES_REQUESTED** [review]

File: packages/core/src/editorUtils/editableInPropertyPage.ts (hunk 1 above)
Comment (alexchuber) [reply packages/core/src/editorUtils/editableInPropertyPage.ts:6]: @sebavan If this same change was made to nodeDecorators.ts in babylon core, would there be any consequences?

Comment (sebavan) [reply packages/core/src/editorUtils/editableInPropertyPage.ts:6]: yup you would need to change the way const enum are transpiled with preserveConstEnums in the ts config I guess.

**Review by AmoebaChant — APPROVED** [review]

**Review by sebavan — APPROVED** [review]

**Review by AmoebaChant — APPROVED** [review]

**Pull request merged**